File: WebApplication/WebApplication/Models/ViewData/RepairOrderViewData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using Maintenance.Models.MaintenanceEntities;

namespace WebApplication.Models.ViewData
{
    public class RepairOrderViewData {
        // id заявки на ремонт
        public int Id { get; set; }
        // дата подачи заявки
        public DateTime DateOfTheApplication { get; set; }
        // статус готовности заявки
        public bool IsReady { get; set; }
        // данные о клиенте
        public ClientViewData ClientViewData { get; set; }
        // данные о автомобиле
        public CarViewData CarViewData { get; set; }
        // данные о работнике
        public WorkerViewData WorkerViewData { get; set; }
        // список неиспраностей
        public List<MalfunctionViewData> MalfunctionViewModels { get; set; }

        public RepairOrderViewData(RepairOrder order, ClientViewData clientViewData, CarViewData carViewData,
            WorkerViewData workerViewData, List<MalfunctionViewData> malfunctionViewModels) {
            Id = order.Id;
            DateOfTheApplication = order.DateOfTheApplication;
            IsReady = order.IsReady;
            ClientViewData = clientViewData;
            CarViewData = carViewData;
            WorkerViewData = workerViewData;
            MalfunctionViewModels = malfunctionViewModels;

            Price = MalfunctionViewModels.Sum(m => m.Price);
            DateOfCompletion = DateTime.Now + TimeSpan.FromHours(MalfunctionViewModels.Sum(m => m.TimeToFix) + 12);
        }

        // стоимость ремонта
        public decimal Price { get; private set; }

        // дата завершения ремонта
        public DateTime DateOfCompletion { get; private set; }
    }
}

File: WebClient/src/models/view-data/repair-order-view-data.ts
import { CarViewData } from './car-view-data';
import { ClientViewData } from './client-view-data';
import { MalfunctionViewData } from './malfunction-view-data';
import { WorkerViewData } from './worker-view-data';

// класс для предоставления заявок на ремонт
export class RepairOrderViewData {
  // конструктор со свойствами класса
  constructor(
    public id: number,  // id - заявки на ремонт
    public dateOfTheApplication: Date, // дата обращения
    public price: number,  // стоимость ремонта
    public dateOfCompletion: Date, // примерная дата завершения
    public isReady: boolean, // выполнен ли ремонт
    public clientViewData: ClientViewData, // данные о клиенте
    public carViewData: CarViewData, // данные о авто
    public workerViewData: WorkerViewData, // данные о работниках
    public malfunctionViewModels: MalfunctionViewData[], // данные о неисправностях
  ) {}
}
